public/pages/CreateHistoricalDetector/components/Configuration/components/ExistingDetectors/ExistingDetectors.tsx

```diff
@@ -0,0 +1,235 @@
+import { CREATE_HISTORICAL_DETECTOR_OPTIONS } from '../../../../utils/constants';
+import { sanitizeSearchText } from '../../../../../utils/helpers';
+import {
+  populateDetectorFieldsFromDetector,
+  populateDetectorFieldsToInitialValues,
+  untouchDetectorFields,
+} from './utils/helpers';
+import { getAllDetectorOptions } from '../../../../utils/helpers';
+import { GetDetectorsQueryParams } from '../../../../../../../server/models/types';
+
+interface ExistingDetectorsProps {
+  formikProps: FormikProps<HistoricalDetectorFormikValues>;
+}
+
+export function ExistingDetectors(props: ExistingDetectorsProps) {
+  const dispatch = useDispatch();
+  const selectedIndexOption = get(
+    props,
+    'formikProps.values.index.0.label',
+    ''
+  );
+  const isIndexSelected = selectedIndexOption && selectedIndexOption.length > 0;
+  const adState = useSelector((state: AppState) => state.ad);
+  const detectors = adState.detectors;
+  const allDetectors = {
+    ...adState.detectorList,
+    ...adState.historicalDetectorList,
+  };
+  const allDetectorOptions = isIndexSelected
+    ? getAllDetectorOptions(Object.values(allDetectors))
+    : [];
+  const [queryText, setQueryText] = useState('');
+  const [selectedDetectorId, setSelectedDetectorId] = useState<string>('');
+  const selectedDetector = detectors[selectedDetectorId];
+  const [createDetectorSelection, setCreateDetectorSelection] = useState<
+    CREATE_HISTORICAL_DETECTOR_OPTIONS
+  >(CREATE_HISTORICAL_DETECTOR_OPTIONS.CREATE_NEW);
+
+  const existingDetectorsOptions = [
+    {
+      id: CREATE_HISTORICAL_DETECTOR_OPTIONS.CREATE_NEW,
+      label: 'No',
+    },
+    {
+      id: CREATE_HISTORICAL_DETECTOR_OPTIONS.USE_EXISTING,
+      label: 'Yes',
+    },
+  ];
+
+  const fetchAllDetectors = async (params: GetDetectorsQueryParams) => {
+    dispatch(getDetectorList(params));
+    dispatch(getHistoricalDetectorList(params));
+  };
+
+  // Fetch all realtime and historical detectors that are using the specified index
+  useEffect(() => {
+    if (isIndexSelected) {
+      const params = {
+        ...GET_ALL_DETECTORS_QUERY_PARAMS,
+        indices: selectedIndexOption,
+      };
+      fetchAllDetectors(params);
+    }
+  }, [selectedIndexOption]);
+
+  // Update the form if a change in selected detector, and get index mappings
+  useEffect(() => {
+    if (selectedDetector) {
+      populateDetectorFieldsFromDetector(props.formikProps, selectedDetector);
+      const indexName = selectedDetector.indices[0];
+      dispatch(getMappings(indexName));
+    } else {
+      populateDetectorFieldsToInitialValues(props.formikProps);
+    }
+  }, [selectedDetector]);
+
+  const getDetectorInfo = (detectorId: string) => {
+    if (detectorId && allDetectors && allDetectors[detectorId]) {
+      dispatch(
+        searchDetector({
+          query: { term: { 'name.keyword': allDetectors[detectorId].name } },
+        })
+      );
+    }
+  };
```

Comment:
It looks like we get detector name by Id, and then search detector by name. Why not use Id directly with [useFetchDetectorInfo](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/public/pages/createDetector/hooks/useFetchDetectorInfo.ts), or just getDetector by Id?

Reply:
Good catch, changed to just call `getDetector` by id.